### ISSUE TITLE
Ainstein US-D1 not working

### ISSUE THREAD

**Issue by msnesbit** [PX4/PX4-Autopilot#19014]
Current build 1.12.3 on a Pixhawk 4
The sens_uland_cfg exists in the parameter list which, as I understand, means that the driver exists on this firmware build.
I have tried every port option I can think of, as well as supplying power to the radar via a BEC and not from the Pixhawk power rail,  but I have yet to read any data from it. The radar is getting power and pulling a very small amount of current which indicates to me that the unit is likely functioning. Any help would be much appreciated!

**Comment by dagar** [comment]
Unfortunately I don't have one of these units, but we can try to debug remotely.

Through a console (mavlink or serial console) can you run `dmesg`, `perf`, and `work_queue status`? I'm trying to see if it's failing with any useful output.

**Comment by msnesbit** [comment]
Here you go. 

![perf2](https://user-images.githubusercontent.com/73359482/149598710-74321775-83a4-4944-91eb-25c7ce363cbc.PNG)
![perf1](https://user-images.githubusercontent.com/73359482/149598712-a21aca3a-4051-4812-ab3f-498152e2634c.PNG)
![dmesg](https://user-images.githubusercontent.com/73359482/149598713-9bfcad61-747a-42cf-a74a-0353b6a3f727.PNG)
![work queue](https://user-images.githubusercontent.com/73359482/149598714-ab0cf38f-269c-4dd5-91d4-4c3a4fec2f79.PNG)

**Comment by FZenjoys** [comment]
@msnesbit Hello, did you solve that problem? I guess I face the same one.
The data output of US D1 is working through its tool test software (see the figure below). Unfortunately, there is no DISTANCE_SENSOR in the MAVLink Inspector after the guide configuration (SENS_ULAND_CFG --> TELEM 2). (power 5.06V, correct RX & TX )
By the way, the TFmini plus is working well via a similar configuration.
 
![output Test screenshot](https://user-images.githubusercontent.com/43603412/201807710-700507ec-58eb-4ccf-93b2-d0d229ae6f52.png)

**Comment by martacportugal** [comment]
> @msnesbit Hello, did you solve that problem? I guess I face the same one. The data output of US D1 is working through its tool test software (see the figure below). Unfortunately, there is no DISTANCE_SENSOR in the MAVLink Inspector after the guide configuration (SENS_ULAND_CFG --> TELEM 2). (power 5.06V, correct RX & TX ) By the way, the TFmini plus is working well via a similar configuration.
> 
> ![output Test screenshot](https://user-images.githubusercontent.com/43603412/201807710-700507ec-58eb-4ccf-93b2-d0d229ae6f52.png)

Hello! :) Did you manage to solve this? I have the same problem.

**Comment by msnesbit** [comment]
Yeah I did end up solving this. The solution is sitting somewhere on an old
email account. Let me dig it up. If I recall, you need to edit one line in
the firmware... easy fix.

**Comment by martacportugal** [comment]
Let me know if you found it! 
I tried adding  'CONFIG_DRIVERS_DISTANCE_SENSOR_ULANDING_RADAR=y' to the default.px4board but it doesn't seem to be working.
